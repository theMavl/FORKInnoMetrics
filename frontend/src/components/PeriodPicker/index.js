import moment from 'moment'
import React from 'react'
import { connect } from 'react-redux'
import _ from 'lodash'
import { Field, isInvalid, reduxForm } from 'redux-form'
import { date, required } from '../../helpers/formValidators'
import Input from '../Input'
import styles from './style.css'

class _PeriodPicker extends React.Component {
  onSubmit = () => {
    console.log('submit')
    this.props.getActivities()
  }

  componentDidUpdate = (prevProps, prevState, snapshot) => {
    if(this.props.periodFormState != undefined && prevProps.periodFormState != undefined &&
       !_.isEqual(this.props.periodFormState.values, prevProps.periodFormState.values)){
      this.props.submit()
    }
}
  render(){
    const datePickerProps = {
      type: 'datePicker',
      labelStyle: {
        display: 'inline',
        padding: 0,
        margin: 0
      },
      placeholder: '',
      containerProps: {style: {
        width: 'fit-content',
        height: '30px'
      }}
    }
    return (
      <div className={styles.periodPicker}>
        <i className={`${'material-icons'} ${styles.calendarIcon}`}>
          calendar_today
        </i>
        <form className={styles.form}
            onSubmit={this.props.handleSubmit(this.onSubmit)}
        >

          <Field
            name='startDate'
            component={Input}
            validate={[required, date]}
            normalize={(value, prevValue, allValues, prevAllValues) => {
              if(moment(value, 'DD/MM/YYYY')
                .isAfter(moment(allValues.endDate, 'DD/MM/YYYY'))
              ){
                this.props.change('endDate', moment(value).format('DD/MM/YYYY'))
                return allValues.endDate
              }
              return moment(value).format('DD/MM/YYYY')
            }}
            props={datePickerProps}
          />
          <span style={{padding: '0 5px'}}>-</span>
          <Field
            name='endDate'
            component={Input}
            validate={[required, date]}
            normalize={(value, prevValue, allValues, prevAllValues) => {
              if(moment(value, 'DD/MM/YYYY')
                .isBefore(moment(allValues.startDate, 'DD/MM/YYYY'))
              ){
                this.props.change('startDate', moment(value).format('DD/MM/YYYY'))
                return allValues.startDate
              }
              return moment(value).format('DD/MM/YYYY')
            }}
            props={datePickerProps}
          />

        </form>
      </div>
    )
  }
}


const initialValues = {
  startDate: moment().subtract({weeks: 1}).format('DD/MM/YYYY'),
  endDate: moment().format('DD/MM/YYYY')
}

let PeriodPicker = reduxForm({
  form: 'periodPicker',
  destroyOnUnmount: false,
  initialValues
})(_PeriodPicker)

PeriodPicker = connect(
  (state) => ({
    periodFormState: state.form.periodPicker,
    submitDisabled: isInvalid('periodPicker')(state) || state.activities.activeRequest
  })
)(PeriodPicker)

export default PeriodPicker