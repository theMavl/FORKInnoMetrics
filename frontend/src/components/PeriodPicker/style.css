.periodPicker {
    /*width: calc(var(--cell-size) * 5);*/
    min-height: var(--cell-size);
    padding: calc(var(--cell-size) / 4);
    background-color: rgba(255, 255, 255, 1);
    box-shadow: 0 0 5px 0px rgba(100, 100, 100, 0.15);

    display: flex;
    align-items: center;
    justify-content: space-between;
    flex-wrap: wrap;
}

.form {
    display: flex;
    align-items: center;
    justify-content: space-between;
    flex-wrap: wrap;
}

.calendarIcon {
    font-size: 30px;
    color: rgba(100, 100, 100, 1);
    padding-right: calc(var(--cell-size) / 4);
}