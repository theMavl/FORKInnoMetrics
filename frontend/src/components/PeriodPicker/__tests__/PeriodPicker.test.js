import React from 'react'
import { Provider } from 'react-redux'
import createStore from 'redux-mock-store'
import { shallow } from 'enzyme/build/index'
import PeriodPicker from '../index'

describe('Test PeriodPicker component', () => {

  it('Test shallow render PeriodPicker component', () => {

    const mockStore = createStore([])
    const store = mockStore({})

    const wrapper = shallow(
      <Provider store={store}>
        <PeriodPicker />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
  })

})