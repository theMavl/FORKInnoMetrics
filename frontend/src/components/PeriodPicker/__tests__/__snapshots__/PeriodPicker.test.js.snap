// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PeriodPicker component Test shallow render PeriodPicker component 1`] = `<Connect(ReduxForm) />`;
